File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let map = {")" : "(", "}": "{", "]" : "["}
    const res = []

    for(letter of s){
        if(letter == "(" || letter == "[" || letter == "{" ){
            res.push(letter)
        }
        else{
            const top = res.pop()
            if(top !== map[letter]) return false
        }
    }

    return res.length == 0
};

//map = {")": "(", "}": "{", "]" : "["}
//if(set.first[]  )

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = { '}' : '{',
               ')': '(', 
               ']':'['
              }
        for char in s:
            if(char in map):
                if(len(stack) > 0 and stack[len(stack) -1] == map[char]):
                    stack.pop()
                else:
                    return False
            else:
                stack.append(char)
        return len(stack) == 0